### PR TITLE
[browser] Adapt bookmark edit dialog to use the new validation style. Contributes to JB#50329

### PR DESCRIPTION
**Pull request by jpetrell**

### CODE REVIEW

File: src/pages/components/BookmarkEditDialog.qml [hunk 1]
```diff
@@ -53,26 +67,21 @@ UserPrompt {
...
-            EnterKey.onClicked: {
-                if (canAccept) {
-                    accept()
-                } else {
-                    titleField.focus = true
-                }
-            }
+            EnterKey.enabled: canAccept
+            EnterKey.iconSource: "image://theme/icon-m-enter-accept"
+            EnterKey.onClicked: accept()
```
Comment (pvuorela) [inline src/pages/components/BookmarkEditDialog.qml:84]: An enter can still be coming from the hardware keyboard. Accept() doesn't seem to protect with canAccept -> all cases like this should be protected?

Comment (pvuorela) [reply src/pages/components/BookmarkEditDialog.qml:84]: Or wait, was there an acceptBlocked happening.

File: src/pages/components/BookmarkEditDialog.qml [hunk 2]
```diff
@@ -53,26 +67,21 @@ UserPrompt {
...
-            placeholderText: qsTrId("sailfish_browser-ph-url_edit")
+            acceptableInput: text.length > 0
+            onActiveFocusChanged: if (!activeFocus) errorHighlight = !acceptableInput
+            onAcceptableInputChanged: if (acceptableInput) errorHighlight = false
 
             //: Label for textfield to edit bookmark's URL
             //% "URL"
             label: qsTrId("sailfish_browser-la-url_editor")
+            //% "URL is required"
+            description:  errorHighlight ? qsTrId("sailfish_browser-la-url_editor_error") : ""
```
Comment (pvuorela) [inline src/pages/components/BookmarkEditDialog.qml:79]: Extra space there.

File: src/pages/components/BookmarkEditDialog.qml [hunk 3]
```diff
@@ -36,15 +46,19 @@ UserPrompt {
...
+
+            acceptableInput: text.length > 0
+            onActiveFocusChanged: if (!activeFocus) errorHighlight = !acceptableInput
+            onAcceptableInputChanged: if (acceptableInput) errorHighlight = false
 
             //: Label for bookmark/favorite's title edit field
             //% "Title"
             label: qsTrId("sailfish_browser-la-title_editor")
+            //% "Title is required"
+            description:  errorHighlight ? qsTrId("sailfish_browser-la-title_editor_error") : ""
```
Comment (pvuorela) [inline src/pages/components/BookmarkEditDialog.qml:61]: Extra space here as well.

**Comment by jpetrell** [comment]
Fixed the empty spaces.

File: src/pages/components/BookmarkEditDialog.qml (hunk 1 above)
Comment (jpetrell) [reply src/pages/components/BookmarkEditDialog.qml:84]: Yeah calling accept() will do same thing as attempting to accept with the dialog header buttons, show error. We would actually need dimmed mode for accept enter key that allowed clicking so could show error highlight there as well.

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**